core/controllers/creator_dashboard_test.py

```diff
@@ -737,21 +738,38 @@ def test_can_update_display_preference(self):
         self.logout()
 
     def test_get_collections_list(self):
-        self.set_admins([self.OWNER_USERNAME])
-        self.login(self.OWNER_EMAIL)
-        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
-        csrf_token = self.get_csrf_token_from_response(response)
-        collection_list = self.get_json(
-            feconf.CREATOR_DASHBOARD_DATA_URL)['collections_list']
-        self.assertEqual(collection_list, [])
-        collection_id = self.post_json(
-            feconf.NEW_COLLECTION_URL, {}, csrf_token=csrf_token)[
-                creator_dashboard.COLLECTION_ID_KEY]
-        collection_list = self.get_json(
-            feconf.CREATOR_DASHBOARD_DATA_URL)['collections_list']
-        self.assertEqual(collection_list[-1]['id'], collection_id)
-        self.assertEqual(len(collection_list), 1)
-        self.logout()
+
+        class MockCollection(object):
```

Comment:
You don't need a MockCollection. That's not necessary here.

Just create a new "standard" collection and specify the correct fields. You should do the research to see what create_default_collection() actually does and what optional arguments it takes.

Reply:
I think the class `MockCollection` is necessary to mock `create_default_collection()` as its a function defined inside the `Collection` class hence it would require a `Collection` instance as its first argument. This can be done by defining the mock function inside a mock collection class

Reply:
@seanlip We want to test the class `NewCollectionHandler` by updating the collection to have some useful data. But the `NewCollectionHandler` class calls `create_default_collection()` without any arguments(i.e the data like title, objective etc has empty values). Therefore we want to mock the function to create a collection with useful data (as you suggested)

Reply:
Actually the purpose of this test is to create a new collection through the `NewCollectionHandler`(by doing a `post_json`) and checking if the collections_list gets updated due to the above request by doing a `get_json` on CreatorDashboardHandler. I named the test `can_create_collections` but renamed it after @nithusha21 suggested [here](https://github.com/oppia/oppia/pull/6480#discussion_r271474398). I am sorry I was not clear before

Reply:
Unit tests should test a single behaviour. Decide whether you're testing NewCollectionHandler or get_collections_list in this test. Then write the test accordingly.

In fact, ideally, you would write separate tests for each of these. They would be different. The first would create a new collection, then retrieve it directly (not via get_collections_list) and verify its fields. The second would populate the datastore with some collections and then call the function to make sure it returns the correct thing.

Reply:
Done